### ISSUE TITLE
FirebaseCore is not a valid configured external dependency

### ISSUE THREAD

**Issue by plu** [tuist/tuist#7119]
### What happened?

The project has a dependency to `Firebase`, and targets like `FirebaseCrashlytics`, `FirebaseCore`, ... can be imported successfully. The project can declare an external dependency to `FirebaseCrashlytics`, but not to `FirebaseCore`.

### How do we reproduce it?

There's a demo project with steps how to reproduce available at [plu/TuistDemos](https://github.com/plu/TuistDemos?tab=readme-ov-file#demo1).

### Error log

```console
$ tuist generate Demo1Framework --no-open
Loading and constructing the graph
It might take a while if the cache is empty
`FirebaseCore` is not a valid configured external dependency
Consider creating an issue using the following link: https://github.com/tuist/tuist/issues/new/choose
```

### macOS version

14.7.1

### Tuist version

4.36.0

### Xcode version

16.1

**Comment by fortmarek** [comment]
Hey @plu 👋 

The resolved `firebase-ios-sdk` version is `11.2.0`. If you take a look at the `Package.swift` for that version, it does _not_ declare `FirebaseCore` as a product: https://github.com/firebase/firebase-ios-sdk/blob/1fc52ab0e172e7c5a961f975a76c2611f4f22852/Package.swift#L27

If you want to use FirebaseCore, you will need to upgrade. The latest version of Firebase does declare `FirebaseCore` as a product: https://github.com/firebase/firebase-ios-sdk/blob/main/Package.swift#L72-L75

**Comment by plu** [comment]
Oh! Thank you, and sorry for not finding that myself :(.
